compendium/DeclarativeServices/test/gtest/TestFactoryTarget.cpp

```diff
@@ -0,0 +1,582 @@
+    {
+        // Register the ServiceB~123 factory service with a mock implementation
+        auto serviceBReg = registerSvc<test::ServiceBInt, MockServiceBImpl>("ServiceB~123", "ServiceBId", "ServiceB~123");
+  
+        // Install and start the bundle containing the ServiceA factory.
+        // DS is now listening for factory configuration objects of the form
+        // ServiceA~<instance>
+        test::InstallLib(context, "TestBundleDSFAC1");
+        cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSFAC1");
+
+        // Create the ServiceA~1 configuration object. Specify a dynamic target
+        // so that the test::ServiceBInt dependency for  ServiceA~1 will only be
+        // satisfied by ServiceB~123 instance.
+        cppmicroservices::AnyMap props;
+        props["test::ServiceBInt"] = std::string("(ServiceBId=ServiceB~123)");
+        auto service = getFactoryService<test::ServiceAInt>("ServiceA", "1", props);
+        ASSERT_TRUE(service);
+
+        // Clean up
+        serviceBReg.Unregister();
+        testBundle.Stop();
+    }
+    /* tFactoryTarget.dependencyExistsAfter.
+     * ServiceA and ServiceB are both factory instances.
+     * ServiceA~1 is dependent on ServiceB~123.
+     * ServiceB~123 is registered after the bundle containing ServiceA is
+     * started.
+     */
+
+    TEST_F(tFactoryTarget, dependencyExistsAfter)
+    {
+        // Install and start the bundle containing the ServiceA factory.
+        // DS is now listening for factory configuration objects of the form
+        // ServiceA~<instance>
+        test::InstallLib(context, "TestBundleDSFAC1");
+        cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSFAC1");
+ 
+        // Create the ServiceA~1 configuration object. Specify a dynamic target
+        // so that the test::ServiceBInt dependency for  ServiceA~1 will only be
+        // satisfied by ServiceB~123 instance.
+        cppmicroservices::AnyMap props;
+        props["test::ServiceBInt"] = std::string("(ServiceBId=ServiceB~123)");
+        auto service = getFactoryService<test::ServiceAInt>("ServiceA", "1", props);
+        // Service cannot be instantiated because it's dependencies are not yet satisfied.
+        ASSERT_TRUE(!service);
+
+        // Create a service tracker for ServiceA. 
+        std::unique_ptr<cppmicroservices::ServiceTracker<test::ServiceAInt>> tracker(new cppmicroservices::ServiceTracker<test::ServiceAInt>(context, nullptr));
+        tracker->Open();
+
+        // Register the ServiceB~123 factory service with a mock implementation
+        auto serviceBReg = registerSvc<test::ServiceBInt, MockServiceBImpl>("ServiceB~123", "ServiceBId", "ServiceB~123");
+   
+        auto serviceA = tracker->WaitForService(std::chrono::milliseconds(500));
+        ASSERT_TRUE(serviceA) << "ServiceA not registered";
+  
+        // Clean up
+        tracker->Close();
+        serviceBReg.Unregister();
+        testBundle.Stop();
+     }
```

Comment:
no return after 177

Reply:
Hmmm. I don't think there should be a return here. 177 marks the end of the TEST_F(tFactoryTarget, dependencyExistsAfter test. It wouldn't compile with a return there.

Reply:
sorry, I meant carriage return. That is my fault. Just need a new line.

Reply:
Will fix